### DISCUSSION TITLE
[Support]: API how to distinguish between Alerts and Detections

### DISCUSSION THREAD

**Discussion by lukicsl** [blakeblackshear/frigate#12574] [General Support] [answered]
### Describe the problem you are having

I have written a python rule in habapp in order to extract incoming requests and create a pushover message when certain criterias are met. I am doing it that way that when I receive a mqtt event message I just extract the vent id which I use call the api to get the details of the event (http://frigate-server:5000/api/events/<evet-id>)
My question now is that I would like to distiguish between alerts, for which I might create an pushover message, and just want to ignore Detections, but do not know which information i need to extract.

which of the fields indicate its just a Detection and not an Alert.
I could filter by zones, though I rather not like to do so.

```
{
  "area": null,
  "box": null,
  "camera": "garage",
  "data": {
    "attributes": [],
    "box": [
      0.9359375,
      0.29444444444444445,
      0.0578125,
      0.25
    ],
    "region": [
      0.5,
      0,
      0.5,
      0.8888888888888888
    ],
    "score": 0.70703125,
    "top_score": 0.7421875,
    "type": "object"
  },
  "detector_type": "edgetpu",
  "end_time": 1721756102.557345,
  "false_positive": null,
  "has_clip": true,
  "has_snapshot": true,
  "id": "1721756053.777208-jd97ij",
  "label": "person",
  "model_hash": "381ddff7e02bc86b7e6c5b1b9cc894db",
  "model_type": "ssd",
  "plus_id": null,
  "ratio": 1,
  "region": null,
  "retain_indefinitely": false,
  "score": null,
  "start_time": 1721756043.777208,
  "sub_label": null,
  "thumbnail": "/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAoHBwgHBgoICAgLCgoLDhgQDg0NDh0VFhEYIx8lJCIfIiEmKzcvJik0KSEiMEEx.....
FgDSqSqkpldvBOKcoKR7t3U8jtSGdlP7v8z1pAf//Z",
  "top_score": null,
  "zones": [
    "strasse"
  ]
}

```

### Version

0.14.0 beta 4

### Frigate config file

```yaml
not relevant
```

### Relevant log output

```shell
not relevant
```

### FFprobe output from your camera

```shell
not relevant
```

### Operating system

Debian

### Install method

Docker Compose

### Object Detector

Coral

### Network connection

Wired

### Camera make and model

reolink, doorbird

**Comment by NickM-27** [comment]
If you want to work off of alerts vs detections then you should use the `/reviews` topic not `/events`

**Reply by lukicsl** [reply]
ok 👍.
looking at the data:
```
{
  "type": "end",
  "before": {
    "id": "1721754795.575895-ubfvks",
    "camera": "front-garden",
    "start_time": 1721754795.575895,
    "end_time": null,
    "severity": "alert",
    "thumb_path": "/media/frigate/clips/review/thumb-front-garden-1721754795.575895-ubfvks.webp",
    "data": {
      "detections": [
        "1721754794.575612-ngnmh2",
        "1721754822.177298-h830pj"
      ],
      "objects": [
        "person"
      ],
      "sub_labels": [],
      "zones": [
        "eingang"
      ],
      "audio": []
    }
  },
  "after": {
...
}
```
Is there any link between review and event, e.g. watching reviews and then from a review find the event.

**Reply by NickM-27** [reply, marked as answer]
data.detections contains all of the event IDs

**Reply by lukicsl** [reply]
Thanks
